arch/arc/Kconfig

```diff
@@ -85,13 +85,35 @@ config ARC_FIRQ
 	  If FIRQ is disabled, the handle of interrupts with highest priority
 	  will be same with other interrupts.
 
-config	ARC_STACK_CHECKING
+config 	ARC_HAS_STACK_CHECKING
+	bool "ARC has STACK_CHECKING"
```

Comment:
Is this a fixed feature of the hardware, or something that it makes sense to enable/disable in the `menuconfig` interface?

If it's a fixed hardware feature, then the symbol should not have a prompt, because then you'll be able to change its value in `menuconfig`. If it's a feature that it makes sense to enable/disable, then the name probably shouldn't have "HAS" in it, because that makes it look like a hardware feature flag.

See [this section](https://docs.zephyrproject.org/latest/guides/kconfig/index.html#what-to-turn-into-kconfig-options) on the Kconfig best practices page.

Reply:
In fact, it's a fixed feature of the hardware.   But because arc is highly configurable,   to avoid creating many different soc in nsim (most options are the same),  it's better to make it configurable from def_config, so we can use different def_config to simulate different soc with one nsim soc configuration. 

What's more, although it's a fixed feature, user can choose whether to use it or not.

Reply:
`Kconfig.defconfig` files might be another option. Might be able to have "true" `_HAS_` symbols that way too, that hide e.g. `ARC_STACK_CHECKING` in the menuconfig when the feature isn't available.

Maybe the help text could mention that the hw needs to have the feature available otherwise.

Anyway, won't block over it. Don't know much about ARC (except it has ties to Starfox on SNES, which is cool :).

Reply:
If the symbol enables use of the HW feature in the code (and assumes it's there), I'd drop the `_HAS_` from the name though.